services/src/main/java/org/keycloak/services/resources/account/AccountCredentialResource.java

```diff
@@ -180,29 +182,41 @@ public CredentialTypeMetadata getMetadata() {
                     .build(session);
             CredentialTypeMetadata metadata = credentialProvider.getCredentialTypeMetadata(ctx);
 
-            List<CredentialRepresentation> userCredentialModels = null;
+            List<CredentialMetadataRepresentation> userCredentialMetadataModels = null;
+
             if (includeUserCredentials) {
-                userCredentialModels = models.stream()
+                List<CredentialModel> modelsOfType = models.stream()
                         .filter(credentialModel -> credentialProvider.getType().equals(credentialModel.getType()))
-                        .map(ModelToRepresentation::toRepresentation)
                         .collect(Collectors.toList());
 
-                if (userCredentialModels.isEmpty() &&
+
+                List<CredentialMetadata> credentialMetadataList = modelsOfType.stream()
+                        .map(m -> {
+                            return credentialProvider.getCredentialMetadata(
+                                    credentialProvider.getCredentialFromModel(m), metadata
```

Comment:
I don't love having to go get the data again here, but I think that's necessary, at least with what I understand right now.

Reply:
I think this is OK to me, at least for now.